crates/codegen/runtime/cargo/crate/src/runtime/parser/parser_support/parser_function.rs

```diff
@@ -57,16 +57,39 @@ where
             ParserResult::PrattOperatorMatch(..) => unreachable!("PrattOperatorMatch is internal"),
 
             ParserResult::NoMatch(no_match) => {
```

Comment:
IIUC, this is a top-level method that will run after the parser is complete, with the `ParserResult` of the entire file/top-most node.
What happens in the case of nested errors, where we recover partially and produce multiple ones (example below)? I don't see a unit test for this case, so it might be a good time to add one.

```solidity
contract A {
  public function // missing the rest...
}

contract B {
  public function // missing the rest...
}
```

Reply:
This is the output, with or without this PR:

```
Tree:
  - (ContractDefinition): # "contract A {\n  public function // missing the rest..." (0..113)
      - (contract_keyword꞉ ContractKeyword): "contract" # (0..8)
      - (leading_trivia꞉ Whitespace): " " # (8..9)
      - (name꞉ Identifier): "A" # (9..10)
      - (leading_trivia꞉ Whitespace): " " # (10..11)
      - (open_brace꞉ OpenBrace): "{" # (11..12)
      - (trailing_trivia꞉ EndOfLine): "\n" # (12..13)
      - (members꞉ ContractMembers): [] # (13..13)
      - (leading_trivia꞉ Whitespace): "  " # (13..15)
      - (UNRECOGNIZED): "public function // missing the rest...\n" # (15..54)
      - (close_brace꞉ CloseBrace): "}" # (54..55)
      - (trailing_trivia꞉ EndOfLine): "\n" # (55..56)
      - (leading_trivia꞉ EndOfLine): "\n" # (56..57)
      - (UNRECOGNIZED): "contract B {\n  public function // missing the rest..." # (57..113)
 ```
 
 What would be the expected outcome?

Reply:
I see. Let me think about it. I can take this offline since it is not blocking.

Reply:
I created issue #1189 for tracking this